retrofit/src/main/java/retrofit2/DeferredRequest.java

```diff
@@ -0,0 +1,13 @@
+package retrofit2;
+
+import com.squareup.okhttp.Request;
+import java.io.IOException;
+
+/**
+ * An un-built HTTP request. This class defers any work necessary to create an HTTP request until
+ * {@link #get()} is called.
+ */
+public interface DeferredRequest {
```

Comment:
I originally used `Callable<Request>` for this, but the fact that it throws a full-on `Exception` was ugly and the naming clashes severely with out `Call` and `Callback` and friends. Really low-value class here. Java 8's `Supplier` doesn't fit either because of the need to throw `IOException` (not that we can use Java 8 types anyway).